www/docs/server/middlewares.md

```diff
@@ -46,3 +46,28 @@ trpc
 :::tip
 See [Error Handling](error-handling.md) to learn more about the `TRPCError` thrown in the above example.
 :::
+
+### Logging
+
+In the example below timings for queries are logged automatically.
+
+```ts
+trpc
+  .router<Context>()
+  .middleware(async ({ path, type, next }) => {
+    const start = Date.now();
+    const result = await next();
+    console.log('request timing:', {path, type, took: Date.now() - start});
```

Comment:
I think the union thing makes this wrong. We now need to check if `ok` is true because otherwise we'll mess up our metrics with fast-throwing errors. Or if not wrong, then unexpected. I think most users would want to think about timing separately from errors.

Reply:
Yeah, docs should probably be updated to reflect this. 


```suggestion
    const durationMs = Date.now() - start;
    res.ok 
      ? console.log('OK request timing:', {path, type, durationMs })
      : console.log('Non-OK request timing', {path, type, durationMs })
    
      
```